tests/mappings/test_sparse_barycenter.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import pytest
+import torch
+
+from fugw.mappings import FUGWSparseBarycenter
+from fugw.utils import _init_mock_distribution
+
+devices = [torch.device("cpu")]
+if torch.cuda.is_available():
+    devices.append(torch.device("cuda:0"))
+
+
+@pytest.mark.parametrize("device", devices)
+def test_fugw_barycenter(device):
+    np.random.seed(0)
+    n_subjects = 4
+    n_voxels = 100
+    n_features = 10
+
+    mesh_sample = np.random.randint(0, n_voxels, size=10)
+
+    # Generate random training data for n subjects
+    features_list = []
+    weights_list = []
+
+    for _ in range(n_subjects):
+        weights, features, geometry_embedding, _ = _init_mock_distribution(
+            n_features, n_voxels
+        )
+        weights_list.append(weights)
+        features_list.append(features)
+
+    fugw_barycenter = FUGWSparseBarycenter(selection_radius=5)
+
+    # Fit the barycenter
+    fugw_barycenter.fit(
+        weights_list,
+        features_list,
+        [geometry_embedding],
+        mesh_sample=mesh_sample,
+        coarse_mapping_solver_params={"nits_bcd": 2, "nits_uot": 10},
+        fine_mapping_solver_params={"nits_bcd": 2, "nits_uot": 10},
+        device=device,
+        verbose=True,
+    )
```

Comment:
This is only a smoke test ?

Reply:
Yes this solely for the purpose of testing that everything runs regardless the validity of the transport plan.

Reply:
Don't we need also a full test ?

Reply:
I added new shape assertions + type checks

Reply:
Maybe we could test edge cases (although this may require using external libraries):
- one could test that `alpha=0` returns wasserstein barycenters similar to that of POT
- similarly, one could test that `alpha=1` returns gromov barycenters similar to that of POT